chia/data_layer/util/plugin.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import List
+
+from chia.data_layer.data_layer_util import PluginRemote
+
+
+async def load_plugin_configurations(root_path: Path, config_type: str) -> List[PluginRemote]:
+    """
+    Loads plugin configurations from the specified directory and validates that the contents
+    are in the expected JSON format (an array of PluginRemote objects). It gracefully handles errors
+    and ensures that the necessary directories exist, creating them if they do not.
+
+    Args:
+        root_path (Path): The root path where the plugins directory is located.
+        config_type (str): The type of plugins to load ('downloaders' or 'uploaders').
+
+    Returns:
+        List[PluginRemote]: A list of valid PluginRemote instances for the specified plugin type.
+    """
+    config_path = root_path / "plugins" / config_type
+    config_path.mkdir(parents=True, exist_ok=True)  # Ensure the config directory exists
+
+    valid_configs = []
+    for conf_file in config_path.glob("*.conf"):
+        try:
+            with open(conf_file) as file:
+                data = json.load(file)
+
+            valid_configs.extend([PluginRemote.unmarshal(marshalled=item) for item in data])
```

Comment:
Will need to take a `log` object as a parameter.

```suggestion
            valid_configs.extend([PluginRemote.unmarshal(marshalled=item) for item in data])
            log.info(f"loaded plugin configuration: {conf_file}")
```